src/extract_coefs.jl

```diff
@@ -0,0 +1,207 @@
+"""
+extract_symbol(t::AbstractTerm) = t.sym
+
+extract_symbol(t::InterceptTerm) = "(Intercept)"
+extract_symbol(t::BSplineTerm) = repeat([t.term.sym], t.df - 1)
+extract_symbol(t::InteractionTerm) = extract_symbol.(t.terms)
+extract_symbol(t::FunctionTerm) = extract_symbol.(t.args)
+
+extract_symbol(t::MatrixTerm) = extract_symbol(t.terms)
+extract_symbol(t::Unfold.TimeExpandedTerm) =
+    repeat(extract_symbol(t.term), inner = length(Unfold.colnames(t.basisfunction)))
+extract_symbol(f::FormulaTerm) = vcat(extract_symbol.(f.rhs)...)
+extract_symbol(t::Vector) = vcat(extract_symbol.(t)...)
+extract_symbol(t::Tuple) = vcat(extract_symbol.(t)...)
+
+"""
+    contained_or_equal(p, e)
+
+Test if `p` equals `e` or whether `e` contains `p` if `e` is a tuple.
+"""
+contained_or_equal(p, e) = (p == e)
+contained_or_equal(p, e::Tuple) = (p in e)
+
+"""
+    get_predictor_string(p)
+
+Return string representation based on the type of `p`.
+
+This function is used for a useful display of variables e.g. in an error message.
+
+# Examples
+```jldoctest
+julia> get_predictor_string(:condition)
+":condition"
+```
+
+```jldoctest
+julia> get_predictor_string("(Intercept)")
+"\"(Intercept)\""
+```
+"""
+get_predictor_string(p::Symbol) = ":$p"
+get_predictor_string(p::String) = "\"$p\""
+get_predictor_string(p::Tuple) = "$p"
+
+"""
+    get_basisnames(model::UnfoldModel)
+
+Return the basisnames for all predictor terms as a vector.
+
+The returned vector contains the name of the event type/basis, repeated by their actual coefficient number (after StatsModels.apply_schema).
+If a model has more than one event type (e.g. stimulus and fixation), the vectors are concatenated.
+"""
+function get_basisnames(model::UnfoldLinearModel)
+    # Extract the event names from the design
+    design_keys = keys(Unfold.design(model))
+
+    # Create a list of the basis names corresponding to each model term
+    basisnames = String[]
+    for (ix, event) in enumerate(design_keys)
+        push!(basisnames, repeat(["event: $(event)"], size(modelmatrix(model)[ix], 2))...)
+    end
+    return basisnames
+end
+
+get_basisnames(model::UnfoldLinearModelContinuousTime) =
+    Unfold.extract_coef_info(Unfold.get_coefnames(model), 1)
+
+"""
+    extract_coefs(model::UnfoldModel, predictor, basisname)
+
+Return the coefficients of an Unfold model for a certain `predictor` and `basisname`.
+
+For extracting the terms of a predictor variable `predictor` must be a symbol e.g. :continuous.
+For extracting the intercept `predictor` should be a String, i.e. "(Intercept)".
+
+`basisname` must match one of the basis names which can be found in `coeftable(model)`.
+
+Note: If a predictor variable has more than one term in the formula (e.g. a spline set, a categorical variable with several levels or an interaction),
+the coefficients for all terms are returned.
+
+The dimensions of the returned coefficients are channel x times x coefficients.
+"""
```

Comment:
@behinger Could you have a short look at the docstrings for the `extract_coefs` function and its methods and tell me whether they make sense or whether you would adapt them?

Reply:
very nice!